cylc/flow/task_pool.py

```diff
@@ -1477,304 +1463,583 @@ def can_spawn(self, name: str, point: 'PointBase') -> bool:
-            self.spawn_on_all_outputs(itask, completed_only=True)
+        if itask is None:
             return None
 
-        LOG.debug(f"[{itask}] spawned")
-        self.db_add_new_flow_rows(itask)
+        # Update it with outputs that were already completed.
+        info = self.workflow_db_mgr.pri_dao.select_task_outputs(
+            itask.tdef.name, str(itask.point))
+        if not info:
+            # (Note still need this if task not run before)
+            self.db_add_new_flow_rows(itask)
+        for outputs_str, fnums in info.items():
+            if flow_nums.intersection(fnums):
+                for msg in json.loads(outputs_str):
+                    itask.state.outputs.set_completed_by_msg(msg)
         return itask
 
-    def force_spawn_children(
+    def _standardise_prereqs(
+        self, prereqs: 'List[str]'
+    ) -> 'Dict[Tokens, str]':
+        """Convert prerequisites to a map of task messages: outputs.
+
+        (So satsify_me logs failures)
+
+        """
+        _prereqs = {}
+        for prereq in prereqs:
+            pre = Tokens(prereq, relative=True)
+            # add implicit "succeeded"; convert "succeed" to "succeeded" etc.
+            output = TaskTrigger.standardise_name(
+                pre['task_sel'] or TASK_OUTPUT_SUCCEEDED)
+            # Convert outputs to task messages.
+            try:
+                msg = self.config.get_taskdef(
+                    pre['task']
+                ).outputs[output][0]
+                cycle = standardise_point_string(pre['cycle'])
+            except KeyError:
+                # The task does not have this output.
+                LOG.warning(
+                    f"output {pre.relative_id_with_selectors} not found")
+                continue
+            except WorkflowConfigError as exc:
+                LOG.warning(
+                    f'Invalid prerequisite task name:\n{exc.args[0]}')
+            except PointParsingError as exc:
+                LOG.warning(
+                    f'Invalid prerequisite cycle point:\n{exc.args[0]}')
+            else:
+                _prereqs[pre.duplicate(task_sel=msg, cycle=cycle)] = prereq
+        return _prereqs
+
+    def _standardise_outputs(
+        self, point: 'PointBase', tdef: 'TaskDef', outputs: List[str]
+    ) -> List[str]:
+        """Convert output names to task output messages."""
+        _outputs = []
+        for out in outputs:
+            # convert "succeed" to "succeeded" etc.
+            output = TaskTrigger.standardise_name(out)
+            try:
+                msg = tdef.outputs[output][0]
+            except KeyError:
+                LOG.warning(f"output {point}/{tdef.name}:{output} not found")
+                continue
+            _outputs.append(msg)
+        return _outputs
+
+    def set_prereqs_and_outputs(
         self,
         items: Iterable[str],
-        outputs: Optional[List[str]] = None,
-        flow_num: Optional[int] = None
+        outputs: List[str],
```

Comment:
May be over-picky. Feel free to ignore.

I'm getting a bit picky, and it will mess up loads of tests at this point, but IMO it would make far more sense if order of prereqs and outputs in the method name and method arguments were consistent. 

`set_outputs_and_prereqs` would be an easier change and would reflect the idea that outputs from one task are prereqs of another and therefore outputs go first.

Reply:
I'm not immune to pickiness myself, to say the least, but I think I'll leave this for now. My "merge finger" is getting twitchy. Besides prerequisites -> outputs -> prerequisites -> outputs ... it's a bit chicken -> egg -> chicken 😁